### PR TITLE
Aktualizr

### PR DESCRIPTION
**Pull request by leon-anavi**
Hi,

The following changes are provided:
* Align with the latest changes from the master branch
* Update Aktualizr to a newer version with cmake file that allows skipping the build of the tests in the Yocto/OE recipe
* Remove gmock Yocto/OE as it is no longer needed by Aktualizr
* Add systemd service for Aktualizr

Best regards,
Leon

### CODE REVIEW

**Comment by gunnarx** [comment]
1. Why is "navit: Remove Navit & all it's dependents" commit part of the PR?  
(Whatever the reason, it has been merged now anyway so you might have to rebase, @leon-anavi)

2. Secondly, the PR builds seem to fail because there is no branch named fix/boost-yocto in the ATS aktualizr repository, and the SRCREV a6136f091 is also missing as far as I can see.
Click on any "details" link for the builds to go to go.genivi.org

File: meta-genivi-dev/recipes-sota/aktualizr/aktualizr_git.bb [hunk 1]
```diff
@@ -12,19 +12,27 @@ S = "${WORKDIR}/git"
...
 SRC_URI = " \
           git://github.com/advancedtelematic/aktualizr;branch=fix/boost-yocto \
+          file://aktualizr.service \
           "
 
+SYSTEMD_PACKAGES = "${PN}"
+SYSTEMD_SERVICE_${PN} = "aktualizr.service"
+
 DEPENDS = "boost curl openssl jansson dbus"
 RDEPENDS_aktualizr = "dbus-lib"
```
Comment (chbae) [inline meta-genivi-dev/recipes-sota/aktualizr/aktualizr_git.bb]: I thought that this is not required.

File: meta-genivi-dev/recipes-sota/aktualizr/aktualizr_git.bb [hunk 2]
```diff
@@ -12,19 +12,27 @@ S = "${WORKDIR}/git"
...
 
 EXTRA_OECMAKE = "-DWARNING_AS_ERROR=OFF -DCMAKE_BUILD_TYPE=Release -DBUILD_TESTS=OFF -DBUILD_GENIVI=ON"
 
+do_install() {
+  install -d ${D}${systemd_unitdir}/system
```
Comment (chbae) [inline meta-genivi-dev/recipes-sota/aktualizr/aktualizr_git.bb]: 1. ${systemd_unitdir}/system -> ${systemd_system_unitdir}
2. I recommend that you use 4 spaces

File: meta-genivi-dev/recipes-sota/aktualizr/aktualizr_git.bb [hunk 3]
```diff
@@ -12,19 +12,27 @@ S = "${WORKDIR}/git"
...
+  install -m 0644 ${WORKDIR}/aktualizr.service ${D}${systemd_unitdir}/system
```
Comment (chbae) [inline meta-genivi-dev/recipes-sota/aktualizr/aktualizr_git.bb]: ${systemd_unitdir}/system -> ${systemd_system_unitdir}

File: meta-genivi-dev/recipes-sota/aktualizr/aktualizr_git.bb [hunk 4]
```diff
@@ -12,19 +12,27 @@ S = "${WORKDIR}/git"
...
+}
+
 FILES_${PN} = " \
               ${bindir}/aktualizr \
               ${libdir}/libjwt.so.0 \
               ${libdir}/libjwt.so.0.3.0 \
+              ${systemd_unitdir}/system/aktualizr.service \
```
Comment (chbae) [inline meta-genivi-dev/recipes-sota/aktualizr/aktualizr_git.bb]: ${systemd_unitdir}/system -> ${systemd_system_unitdir}

File: meta-genivi-dev/recipes-sota/aktualizr/aktualizr_git.bb [hunk 5]
```diff
@@ -12,19 +12,27 @@ S = "${WORKDIR}/git"
...
 
+do_install() {
+  install -d ${D}${systemd_unitdir}/system
+  install -m 0644 ${WORKDIR}/aktualizr.service ${D}${systemd_unitdir}/system
+}
+
 FILES_${PN} = " \
               ${bindir}/aktualizr \
               ${libdir}/libjwt.so.0 \
               ${libdir}/libjwt.so.0.3.0 \
```
Comment (chbae) [inline meta-genivi-dev/recipes-sota/aktualizr/aktualizr_git.bb]: those 3 lines are really required?

**Review by chbae — CHANGES_REQUESTED** [review]
Could you rebase on current master? and why don't you remove merge commit?

**Comment by leon-anavi** [comment]
Hi @gunnarx, @chbae,

Thank you for the provided feedback. I am on my way to ELC therefore I will not be able to address these issues immediately. The "navit: Remove Navit & all it's dependents" commit is part of the PR because I have merged the latest changes from master in this branch. As @chbae suggested I will rebase again and remove the merge commit. Yesterday I emailed ATS Advanced Telematic Systems asking about branch fix/boost-yocto. I will fix and provide an update of the recipes as soon as I return from ELC.

Best regards,
Leon

**Comment by chbae** [comment]
@leon-anavi Cool. enjoy your trip :)

**Comment by zeenix** [comment]
@leon-anavi I hope you had a great trip and talk! I just wanted to remind you about this PR.

**Comment by leon-anavi** [comment]
Hi,

ATS Advanced Telematic Systems informed me that Aktualizr is ready for upstreaming in meta-genivi-dev master branch therefore I will close this GitHub pull request, address the issues pointed out by @gunnarx, @chbae, @zeenix, get rid of some redundant patches (like the one for gmock) and rebase just the needed patches on top of master.

Best regards,
Leon

**Comment by zeenix** [comment]
That's great to hear @leon-anavi ! I'd highly recommend reusing this PR (if possible) but no biggie if you don't.

**Comment by cajun-rat** [comment]
@zeenix Is there a reason why we shouldn't be merging to master--what would be the benefit of merging from our branch, then to a branch in GENIVI, then to GENIVI master?

**Comment by zeenix** [comment]
@cajun-rat Actually we prefer merging to master always. Please read the thread here for reasons on why it was done this way: https://github.com/GENIVI/meta-genivi-dev/pull/58

**Comment by zeenix** [comment]
@cajun-rat @leon-anavi Although now that I think more about it, I don't see the point of having a branch for a contributor in the mainline repository if the contributor in question doesn't have write-access to the repo. Since one can always create a clone with 1 click and then also very easily create a PR from there, I think we shouldn't do this in future.

File: meta-genivi-dev/recipes-sota/aktualizr/aktualizr_git.bb (hunk 3 above)
Comment (leon-anavi) [reply meta-genivi-dev/recipes-sota/aktualizr/aktualizr_git.bb]: OK, thank you. I have fixed it.

File: meta-genivi-dev/recipes-sota/aktualizr/aktualizr_git.bb (hunk 2 above)
Comment (leon-anavi) [reply meta-genivi-dev/recipes-sota/aktualizr/aktualizr_git.bb]: OK, thank you.

File: meta-genivi-dev/recipes-sota/aktualizr/aktualizr_git.bb (hunk 4 above)
Comment (leon-anavi) [reply meta-genivi-dev/recipes-sota/aktualizr/aktualizr_git.bb]: OK, thank you.

File: meta-genivi-dev/recipes-sota/aktualizr/aktualizr_git.bb (hunk 1 above)
Comment (leon-anavi) [reply meta-genivi-dev/recipes-sota/aktualizr/aktualizr_git.bb]: Yes, according to the [mega manual ](http://www.yoctoproject.org/docs/latest/ref-manual/ref-manual.html#ref-classes-systemd) there is no need of SYSTEMD_PACKAGES if the service is in the main package. The name of the service file should be in SYSTEMD_SERVICE.

**Comment by cajun-rat** [comment]
@zeenix Yes, that makes sense

**Pull request closed without merging**